File: 252-meeting-rooms/meeting-rooms.js
/**
 * @param {number[][]} intervals
 * @return {boolean}
 */
var canAttendMeetings = function(intervals) {
    if (!intervals || intervals.length === 0) return true

    intervals.sort((a, b) => a[0] - b[0])

    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i-1][1] > intervals[i][0]) {
            return false
        }
    }

    return true
};

File: 252-meeting-rooms/meeting-rooms.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        if not intervals or len(intervals) == 0:
            return True

        intervals.sort()

        for i in range(1, len(intervals)):
            if intervals[i-1][1] > intervals[i][0]:
                return False

        return True